lib/OpenQA/Task/Job/ArchiveResults.pm

```diff
@@ -0,0 +1,43 @@
+# Copyright (C) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Task::Job::ArchiveResults;
+use Mojo::Base 'Mojolicious::Plugin', -signatures;
+
+sub register ($self, $app, @args) {
+    $app->minion->add_task(archive_job_results => \&_archive_results);
+}
+
+sub _archive_results ($minion_job, @args) {
+    my ($openqa_job_id) = @args;
+    my $app = $minion_job->app;
+    return $minion_job->fail('No job ID specified.') unless defined $openqa_job_id;
+
+    # FIXME: Should we ensure that the job results are finalized?
+    # FIXME: Should we take care of possible interference with cleanup task?
+    # FIXME: This breaks jobs currently loading the results tab. I suppose we have to live
+    #        with it.
```

Comment:
good point. I wonder how that would break and if we can intervene within the results tab loading to force-refresh loading that page?

Reply:
I also wonder how that would break. Only HTTP requests which are pending when deleting the old results directory and had already been started before the job was marked as archived would cause problems.

We have controller code reading result files and some files are served directly by Apache. Likely the deletion of the original files during the archiving would fail if the result file if there are still file handles open for reading at this point. So we would likely see failed Minion jobs and leftover results. We could make it more resilient against that problem by trying to delete the results again after a short delay.

If it was actually possible to delete the files while there are still file handles open for reading I suppose the readers would run into I/O errors and at least some parts of the details tab wouldn't be shown correctly.